CDCgov/prime-reportstream
# #3096: Append the App Insights key to the appropriate .env file

This PR transfers the GitHub secret with the AppInsights instrumentation key for the appropriate environment. We update the environment-specific `.env` file with the key/value pair so the build of the React app can configure the App Insights plugin from there.

Note that the build of the static site *does not* include any `.env` file in the repo.

Test Steps:
1. Push this change to `master`
2. Browse around some web pages
3. See if there are errors in `console.log`
4. See if there is data in AppInsights

## Changes
* Adds a run step to the action for building the front-end. This step appends the appropriate AppInsights instrumentation key to the appropriate `.env` file for use by the static site build.

## Checklist

### Testing
* Can only test with GH Actions

### Process

## Fixes
- #3096 

## To Be Done

N/A

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
